source/codegen/common_helpers.py

```diff
@@ -328,23 +328,147 @@ def filter_parameters_for_grpc_fields(parameters):
   return [p for p in parameters if p.get('include_in_proto', True)]
 
 
+class AttributeGroup:
+    def __init__(self, name, attributes, config):
+        self.name = name
+        self.attributes = attributes
+        self._config = config
+
+
```

Comment:
Note: I've started working on `Reset`. That will not be a significant change to the **logic** of this code, but it will change a lot of the terminology.

We'll also need to split out `resettable` attributes into another group that will get its own enum. So, code on that level will be a little more generic (I'm thinking "sub_group"). Then when we're expanding accessors, resetters will be handled separately from typed getters/setters. But will use most of the same patch logic.